compiler/damlc/daml-package-config/src/DA/Daml/Package/Config.hs

```diff
@@ -87,7 +87,7 @@ checkPkgConfig PackageConfigFields {pName, pVersion} =
       , "use them as dependencies in other projects. Unsupported package names or versions may"
       , "start causing compilation errors without warning."
       ]
-    versionRegex = "^(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*))?)?$" :: T.Text
+    versionRegex = "^(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*))*$" :: T.Text
```

Comment:
This in principle also allows `1.2`, right? Is that on purpose?

Reply:
Or even `1`.

Could probably change that `*` to `{2,}` if we insist on at least 3 components.

Reply:
Ah, right, thanks. Reading regexps is difficult. Thanks for pointing that out.

Reply:
GHC is happy with that and I don’t really see a restriction to restrict it further if we have to allow 5 component names.
Ideally, I’d enforce semver but I can’t because I tried to patch GHC to support that and it resulted in ghci segfaulting in the linker and I gave up.